### DISCUSSION TITLE
Reformatting

### DISCUSSION THREAD

**Discussion by DevSriAkash** [vivian-dai/PicoCTF2021-Writeup#13] [Q&A] [answered]
Hey there! I didn't understand the reformatting of the cipher we got. I understand the removal of (nil), &, + and = but I don't understand how the order changed. Can you help me out?

**Comment by vivian-dai** [comment, marked as answer]
The order is different because of [little endian](https://en.wikipedia.org/wiki/Endianness). Little end stores the least significant byte first. We're more used to a big endian format where the most significant byte is stored first.

**Reply by DevSriAkash** [reply]
Thank you for the instant reply mate. Is it hard to convert text from little endian to big endian?

**Reply by vivian-dai** [reply]
The only difference between little and big endian is the order the bytes appear in. For the Stonks question, for every block of 4 characters, the order is reversed. If it was 12345678 in big endian, it would be 43218765 in little endian. Notice it's just reversing the sequence of bytes every set interval

**Reply by DevSriAkash** [reply]
Oh thanks a lot mate. Helped me a lot :D
